### DISCUSSION TITLE
Read filter hook is missing accountability in context

### DISCUSSION THREAD

**Discussion by sensedrive** [directus/directus#15446] [Archive]
### Describe the Bug

I am writing a hook, which should deny any reading permissions, when a certain flag is set at the user, which is trying to read data. I was starting with a very simple code example, to start the development.

The code is the following:

```javascript

import { defineHook } from '@directus/extensions-sdk';

export default defineHook(({ filter }, { accountability }) => {
  filter('items.read', (query, collection) => {
    console.log(accountability); # Undefined
    console.log('Reading Item!');
  });
});

```

### To Reproduce

Execute the code, provided in description and debug.

### Errors Shown

1.  According to the docs the context object should contain **accountability**. https://docs.directus.io/extensions/hooks.html#filter
2. The hook is only called on user collections, not on system collections.

**Comment by br41nslug** [comment]
> 1. According to the docs the context object should contain accountability. https://docs.directus.io/extensions/hooks.html#filter

The context object this is referring to is for the filter event itself not the top-level extension context
```js
import { defineHook } from '@directus/extensions-sdk';

export default defineHook(({ filter }, { }) => {
  filter('items.read', (input, meta, context) => {
    const { accountability } = context;
    console.log('Reading Item!');
  });
});
```
> 2. The hook is only called on user collections, not on system collections.

Event for the system collections require you to specify the collection explicitly `<system-collection>.create` https://docs.directus.io/extensions/hooks.html#available-events

**Reply by sensedrive** [reply]
Thank you for having a quick and immediate look at this!
